src/rust/src/http_reader.rs

```diff
@@ -145,14 +151,18 @@ impl HttpFgbReader {
             self.pos = self.feature_base + item.offset;
         }
         self.feat_no += 1;
-        let bytes = self.client.get_range(self.pos, 4, min_req_size).await?;
+        let mut bytes = BytesMut::from(self.client.get_range(self.pos, 4, min_req_size).await?);
         self.pos += 4;
-        let feature_size = LittleEndian::read_u32(bytes) as usize;
-        let bytes = self
-            .client
-            .get_range(self.pos, feature_size, min_req_size)
-            .await?;
+        let feature_size = LittleEndian::read_u32(&bytes) as usize;
+        bytes.put(
+            self.client
+                .get_range(self.pos, feature_size, min_req_size)
+                .await?,
+        );
         self.fbs.feature_buf = bytes.to_vec(); // Not zero-copy
+                                               // verify flatbuffer
```

Comment:
```suggestion
        // verify flatbuffer
```

nit: I think this comment applies to the subsequent line, so this is a weird indent 🤓

Reply:
I blaim `cargo fmt` for that....